File: C++/stack/stack.cpp
#include "stack.h"
using namespace std;

Stack::Stack()
{
    top=-1;
    for (int i=0;i<MAX_STACK_SIZE;i++)
    {
        this->stack_array[i]=-1;
    }
}

Stack::~Stack(){}

bool Stack::isEmpty()
{
   return (top<0);
}

bool Stack::isFull()
{
    return (top>=MAX_STACK_SIZE-1);
}

void Stack::push(int val)
{
    if (!isFull())
    {
        top++;
        this->stack_array[top]=val;
    }
    else
    {
        cout<<"Stack is Full. Stack Overflow."<<endl;
    }
}

int Stack::pop()
{
    if (!isEmpty())
    {
        return this->stack_array[top--];
    }
    else
    {
        cout<<"Stack underflow."<<endl;
        return -1;
    }
}

int Stack::peek()
{
    if (!isEmpty())
    {
        return this->stack_array[top];
    }
    else
    {
        cout<<"Stack underflow."<<endl;
        return -1;
    }
}

void Stack::traverse()
{
    for (int i=0;i<MAX_STACK_SIZE;i++)
    {
        if (this->stack_array[i]==-1)
        {
            break;
        }
        else
        {
            cout<<this->stack_array[i]<<endl;
        }
    }
}

File: Python/stack/stack.py
class Stack():
    def __init__(self, stack_size):
        self.stack_size = stack_size
        self.elements = []

    def push(self, ele):
        if not self.is_full:
            self.elements.append(ele)
        else:
            print("Stack Overflow!")

    def pop(self):
        if not self.is_empty:
            return self.elements.pop()
        else:
            print("Stack Underflow!")

    def top(self):
        if not self.is_empty:
            return self.elements[-1]
        else:
            print("Stack Underflow!")
    
    @property
    def is_empty(self):
        return len(self.elements) == 0

    @property
    def is_full(self):
        return len(self.elements) == self.stack_size
